schema/validator.go

```diff
@@ -26,14 +26,42 @@ import (
 	"github.com/xeipuuv/gojsonschema"
 )
 
+type validateRefMediaFunc func(r io.Reader) error
+
+var vlidateRefMediaMap = map[Validator]validateRefMediaFunc{
+	ValidatorMediaTypeImageIndex: validateImageIndexRefObject,
+	ValidatorMediaTypeManifest:   validateManifestRefObject,
+}
+
+// ValidateFunc provides the type of function to validate specific object.
+type ValidateFunc func(r io.Reader, v Validator) error
```

Comment:
If we expect to be applying a series of these to validate a single blob, using an `io.Reader` seems wasteful.  Is there some Go type magic we can use to pass a parsed JSON object through to the validator to let us only deserialize the JSON once?